monitor/utils/events.js

```diff
@@ -164,50 +162,6 @@ async function main(mode) {
       // Get token swap events emitted by foreign bridge
       const bridgeTokensSwappedEvents = tokensSwappedEvents.filter(e => e.address === COMMON_FOREIGN_BRIDGE_ADDRESS)
 
-      // Get transfer events for each previous erc20
-      const uniqueTokenAddressesSet = new Set(bridgeTokensSwappedEvents.map(e => e.returnValues.from))
```

Comment:
We struggled into the same issue two months ago. Removing this piece of code (lines 167-210) breaks the ability of a monitor to handle all transfer events from the initial block. I have just run the monitor, here is what it shows:
```json
{
    "home": {
        "totalSupply": "56148.485124074190532071",
        "deposits": 323,
        "withdrawals": 850
    },
    "foreign": {
        "erc20Balance": "144",
        "investedErc20Balance": "56008.373383986565701527",
        "accumulatedInterest": "0.061364733920728899",
        "deposits": 323,
        "withdrawals": 784
    },
    "balanceDiff": 3.8882599123751693,
    "lastChecked": 1596078744,
    "depositsDiff": 0,
    "withdrawalDiff": 66,
    "health": true
}
```
Such thing happens since old Sai transfers are not tracked anymore, however, all corresponding xDai mint events are still present. So the monitor is unable to understand where these xDai coins came from.

So I returned back this piece of code in #335.

Reply:
Thanks for pointing this out! I reverted the changes for the monitor